File: sorting/buble_sort/buble_sort.js
const buble_sort = (arr) => {
  for (let i = 0; i < arr.length - 1; i++) {
    console.log("i= ", i);
    for (let j = 0; j < arr.length - i - 1; j++) {
      console.log("j= ", j);
      if (arr[j] > arr[j + 1]) {
        const temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
      }
    }
  }
};

const arr = [2, 7, 3, 5, 1, 4, 6];

buble_sort_good_variant(arr);

console.log(arr);

File: sorting/buble_sort/buble_sort.py
def buble_sort(arr):
    for i in range(len(arr) - 1):
        for j in range(len(arr) - i - 1):
            if arr[j] > arr[j + 1]:
                # fast swap
                # arr[j], arr[j + 1] = arr[j + 1], arr[j]  
                
                # slow swap
                temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp
                

arr = [1, 5, 3, 2, 4]
buble_sort(arr)
print(arr)